rake_nltk/rake.py

```diff
@@ -166,20 +166,12 @@ def _generate_phrases(self, sentences):
            @return: List of List of strings where each sublist is a collection
            of words which form the contender phrase.
         '''
-        phrase_list = []
+        phrase_list = set()
         # Create contender phrases from sentences.
         for sentence in sentences:
             word_list = [word.lower() for word in wordpunct_tokenize(sentence)]
-            phrase_list.extend(self._get_phrase_list_from_words(word_list))
-        # Drop duplicates from the list.
-        unique_phrase_list = []
-        seen = defaultdict(lambda: False)
-        for phrase in phrase_list:
-            phrase_str = ' '.join(phrase)
-            if not seen[phrase_str]:
-                seen[phrase_str] = True
-                unique_phrase_list.append(phrase)
-        return unique_phrase_list
+            phrase_list.update(self._get_phrase_list_from_words(word_list))
+        return phrase_list
 
     def _get_phrase_list_from_words(self, word_list):
         '''Method to create contender phrases from the list of words that form a
```

Comment:
Please change
`List of phrases: [['red', 'apples'], ['good'], ['flavour']]`
in the function definition to 
`List of phrases: [('red', 'apples'), ('good'), ('flavour')]`
to reflect the change.